utils/core_maths/src/lib.rs

```diff
@@ -0,0 +1,1142 @@
+    }
+
+    #[inline]
+    fn powi(self, mut exp: i32) -> Self {
+        if exp == 0 {
+            return 1.0;
+        }
+        let mut base = self;
+        let mut acc = 1.0;
+
+        while exp > 1 {
+            if (exp & 1) == 1 {
+                acc *= base;
+            }
+            exp /= 2;
+            base = base * base;
+        }
+
+        // since exp!=0, finally the exp must be 1.
+        // Deal with the final bit of the exponent separately, since
+        // squaring the base afterwards is not necessary and may cause a
+        // needless overflow.
+        acc * base
+    }
+
+    #[inline]
+    fn powf(self, n: Self) -> Self {
+        libm::powf(self, n)
+    }
+
+    #[inline]
+    fn sqrt(self) -> Self {
+        libm::sqrtf(self)
+    }
+
+    #[inline]
+    fn exp(self) -> Self {
+        libm::expf(self)
+    }
+
+    #[inline]
+    fn exp2(self) -> Self {
+        libm::exp2f(self)
+    }
+
+    #[inline]
+    fn ln(self) -> Self {
+        libm::logf(self)
+    }
+
+    #[inline]
+    fn log(self, base: Self) -> Self {
+        self.ln() / base.ln()
+    }
+
+    #[inline]
+    fn log2(self) -> Self {
+        libm::log2f(self)
+    }
+
+    #[inline]
+    fn log10(self) -> Self {
+        libm::log10f(self)
+    }
+
+    #[inline]
+    fn cbrt(self) -> Self {
+        libm::cbrtf(self)
+    }
+
+    #[inline]
+    fn hypot(self, other: Self) -> Self {
+        libm::hypotf(self, other)
+    }
+
+    #[inline]
+    fn sin(self) -> Self {
+        libm::sinf(self)
+    }
+
+    #[inline]
+    fn cos(self) -> Self {
+        libm::cosf(self)
+    }
+
+    #[inline]
+    fn tan(self) -> Self {
+        libm::tanf(self)
+    }
+
+    #[inline]
+    fn asin(self) -> Self {
+        libm::asinf(self)
+    }
+
+    #[inline]
+    fn acos(self) -> Self {
+        libm::acosf(self)
+    }
+
+    #[inline]
+    fn atan(self) -> Self {
+        libm::atanf(self)
+    }
+
+    #[inline]
+    fn atan2(self, other: Self) -> Self {
+        libm::atan2f(self, other)
+    }
+
+    #[inline]
+    fn exp_m1(self) -> Self {
+        libm::expm1f(self)
+    }
+
+    #[inline]
+    fn ln_1p(self) -> Self {
+        libm::log1pf(self)
+    }
+
+    #[inline]
+    fn sinh(self) -> Self {
+        libm::sinhf(self)
+    }
+
+    #[inline]
+    fn cosh(self) -> Self {
+        libm::coshf(self)
+    }
+
+    #[inline]
+    fn tanh(self) -> Self {
+        libm::tanhf(self)
+    }
+
+    #[inline]
+    fn asinh(self) -> Self {
+        let ax = self.abs();
+        let ix = 1.0 / ax;
+        (ax + (ax / (Self::hypot(1.0, ix) + ix)))
+            .ln_1p()
+            .copysign(self)
+    }
+
+    #[inline]
+    fn acosh(self) -> Self {
+        if self < 1.0 {
+            Self::NAN
+        } else {
+            (self + ((self - 1.0).sqrt() * (self + 1.0).sqrt())).ln()
+        }
+    }
+
+    #[inline]
+    fn atanh(self) -> Self {
+        0.5 * ((2.0 * self) / (1.0 - self)).ln_1p()
+    }
+}
+
+impl CoreFloat for f64 {
```

Comment:
Issue: Shouldn't you be gating this impl on `feature = "std"` or something?

Reply:
what would that achieve? the idea of this whole crate is to make these methods available without std.